script/DeployCanonGuard.s.sol

```diff
@@ -120,15 +120,29 @@ contract DeployCanonGuard is Constants, Script {
   }
 
   function _deployAllChainsFactories() internal {
-    canonGuardFactory = new CanonGuardFactory{salt: SALT}(address(MULTI_SEND_CALL_ONLY));
-    allowanceClaimorFactory = new AllowanceClaimorFactory{salt: SALT}();
-    preApproveActionFactory = new PreApproveActionFactory{salt: SALT}();
-    cappedTokenTransfersHubFactory = new CappedTokenTransfersHubFactory{salt: SALT}();
-    changeSafeGuardActionFactory = new ChangeSafeGuardActionFactory{salt: SALT}();
-    setEmergencyCallerActionFactory = new SetEmergencyCallerActionFactory{salt: SALT}();
-    setEmergencyTriggerActionFactory = new SetEmergencyTriggerActionFactory{salt: SALT}();
-    simpleActionsFactory = new SimpleActionsFactory{salt: SALT}();
-    simpleTransfersFactory = new SimpleTransfersFactory{salt: SALT}();
+    canonGuardFactory = ICanonGuardFactory(
+      CREATE_X.deployCreate3(
+        bytes32(abi.encodePacked(msg.sender, false, bytes11(SALT))),
+        abi.encodePacked(type(CanonGuardFactory).creationCode, abi.encode(MULTI_SEND_CALL_ONLY))
```

Comment:
hardcoding `MULTI_SEND_CALL_ONLY` seems very unsafe with create3, I expanded a bit in the channel